contracts/roles/AgentManager.sol

```diff
@@ -0,0 +1,112 @@
+pragma solidity ^0.6.0;
+
+import "../token/IToken.sol";
+import "../registry/IIdentityRegistry.sol";
+import "./AgentRoles.sol";
+import "@onchain-id/solidity/contracts/IIdentity.sol";
+
+contract AgentManager is AgentRoles {
+
+    IToken public token;
+    IIdentityRegistry public identityRegistry;
+
+    constructor (address _token) public {
+        token = IToken(_token);
+        identityRegistry = token.getIdentityRegistry();
```

Comment:
I think we can remove the state variable for identity registry and retrieve it each time from the token contract.